be/src/exprs/jit/jit_wrapper.cpp

```diff
@@ -0,0 +1,226 @@
+        LOG(ERROR) << "JIT: Failed to select target machine";
+        return Status::JitCompileError("Failed to select target machine");
+    }
+    _data_layout = std::make_unique<const llvm::DataLayout>(_target_machine->createDataLayout());
+
+    // Create a JIT engine instance.
+    auto jit = llvm::orc::LLJITBuilder().create();
+    if (!jit) {
+        std::string error_message;
+        llvm::handleAllErrors(jit.takeError(), [&](const llvm::ErrorInfoBase& EIB) { error_message = EIB.message(); });
+        LOG(ERROR) << "JIT: Failed to create LLJIT instance" << error_message;
+        return Status::JitCompileError("Failed to create LLJIT instance");
+    }
+    _jit = std::move(jit.get());
+
+    // Initialize pass manager for IR optimization.
+    // TODO(Yueyang): check optimization level.
+    // TODO(Yueyang): check if we need to add more optimization passes.
+    _pass_manager_builder.OptLevel = 3;
+    _pass_manager_builder.SLPVectorize = true;
+    _pass_manager_builder.LoopVectorize = true;
+    _pass_manager_builder.VerifyInput = true;
+    _pass_manager_builder.VerifyOutput = true;
+    _pass_manager_builder.populateModulePassManager(_pass_manager);
+
+    _initialized = true;
+    return Status::OK();
+}
+
+StatusOr<JITScalarFunction> JITWapper::compile_scalar_function(ExprContext* context, Expr* expr) {
+    auto* instance = JITWapper::get_instance();
+    if (!instance->initialized()) {
+        return Status::JitCompileError("JIT engine is not initialized");
+    }
+
+    // TODO(Yueyang): optimize module name.
+    auto expr_name = expr->debug_string();
+
+    // TODO(Yueyang): loopup in cache.
+
+    auto llvm_context = std::make_unique<llvm::LLVMContext>();
+    auto module = std::make_unique<llvm::Module>(expr_name, *llvm_context);
+    instance->setup_module(module.get());
+    // Generate scalar function IR.
+    RETURN_IF_ERROR(JITFunction::generate_scalar_function_ir(context, *module, expr));
+    if (llvm::verifyModule(*module)) {
+        return Status::JitCompileError("Failed to generate scalar function IR");
+    }
+
+    // Optimize module.
+    instance->optimize_module(module.get());
+    if (llvm::verifyModule(*module)) {
+        return Status::JitCompileError("Failed to optimize scalar function IR");
+    }
+
+    // Compile module, return function pointer (maybe nullptr).
+    JITScalarFunction compiled_function = reinterpret_cast<JITScalarFunction>(
+            instance->compile_module(std::move(module), std::move(llvm_context), expr_name));
+
+    if (compiled_function == nullptr) {
+        return Status::JitCompileError("Failed to compile scalar function");
+    }
+
+    // TODO(Yueyang): add to cache.
+
+    // Return function pointer.
+    return compiled_function;
+}
+
+Status JITWapper::remove_function(const std::string& expr_name) {
+    auto* instance = JITWapper::get_instance();
+    if (!instance->initialized()) {
+        return Status::JitCompileError("JIT engine is not initialized");
```

Comment:
named JITEngine is better than JITWapper?

Reply:
Done.